astroimtools/cutout_tools.py

```diff
@@ -16,11 +16,267 @@
+    """
+    Make cutouts from a 2D image and write them to FITS files.
+
+    The input Catalog must have the following columns with unit info, where applicable:
+
+        * ``'id'`` - ID string; no unit necessary.
+        * ``'ra'`` or ``'x'``- RA (e.g., in degrees) or pixel x position (in units.pix).
+        * ``'dec'`` or ``'y'`` - DEC (e.g., in degrees) or pixel y position (in units.pix).
+        * ``'cutout_width'`` - Cutout width (e.g., in arcsec, pix).
+        * ``'cutout_height'`` - Cutout height (e.g., in arcsec, pix).
+    Optional columns:
+        * ``'cutout_pa'`` - Cutout angle (e.g., in degrees). This is only
+          use if user chooses to rotate the cutouts. Positive value
+          will result in a clockwise rotation.
+
+    If saved to fits, cutouts are organized as follows:
+        <output_dir>/
+            <id>.fits
+
+    Each cutout image is a simple single-extension FITS with updated WCS.
+    Its header has the following special keywords:
+
+        * ``OBJ_RA`` - RA of the cutout object in degrees.
+        * ``OBJ_DEC`` - DEC of the cutout object in degrees.
+        * ``OBJ_ROT`` - Rotation of cutout object in degrees.
+
+    Examples
+    --------
+    If you have a list of hubble ultra deep field ra and dec coords,
+    you may use the tool as follows:
+        >>> from astropy.table import Table
+        >>> import astropy.units as u
+
+        >>> ra = [53.18782913, 53.14794797, 53.15059559] * u.deg
+        >>> dec = [-27.79405589, -27.77392421, -27.77158621] * u.deg
+        >>> ids = ["Galax_0", 123, 53.15059559*u.deg]
+        >>> cutout_width = cutout_height = [3.0, 4.0, 3.0] * u.arcsec
+
+        >>> catalog = Table(data=[ids, ra, dec, cutout_width, cutout_height], \
+                            names=['id', 'ra', 'dec', 'cutout_width', 'cutout_height'])
+    To get a list of PrimaryHDU objects:
+        >>> cutouts = cutout_tool('h_udf_wfc_b_drz_img.fits', catalog)
+    To save to fits file, (returns a list of file names):
+        >>> cutouts = cutout_tool('h_udf_wfc_b_drz_img.fits', catalog, to_fits=True)
+    If the above catalog table is saved in an ECSV file with the proper units information:
+        >>> cutouts = cutout_tool('h_udf_wfc_b_drz_img.fits', 'catalog.ecsv')
+
+    Parameters
+    ----------
+    image : str or `HDUList` or `PrimaryHDU` or `ImageHDU` or `CompImageHDU`
+        Image to cut from. If string is provided, it is assumed to be a
+        fits file path.
+    catalog : str or `astropy.table.table.Table`
+        Catalog table defining the sources to cut out. Must contain
+        unit information as the cutouttool does not assume default units.
+    image_ext : int, optional
+        Image extension to extract header and data. Default is 0.
+    origin : int
+        Whether pixel coordinates are 0 or 1-basedpixel coordinates.
+    to_fits : bool
+        Save cutouts to fits files. Each cutout will be saved to a separate file.
+    output_dir : str
```

Comment:
You might want to state if the output_dir needs to exist, or will it create it if it does not exist.